### DISCUSSION TITLE
How does bun set up JavaScriptCore to run the entry point script file contents?

### DISCUSSION THREAD

**Discussion by RaisinTen** [oven-sh/bun#2398] [Q&A] [answered]
When I do `bun run script.js`, how does bun set up JavaScriptCore for executing the contents of this file? Can someone please help me find the exact location in this repo where this is done? I was looking for a call to [evaluateScript](https://developer.apple.com/documentation/javascriptcore/jscontext/1451384-evaluatescript?language=objc) but I wasn't able to find any.

**Comment by Jarred-Sumner** [comment, marked as answer]
This function is where the script execution context is initialized: https://github.com/oven-sh/bun/blob/6f59bc7d2e7aa64956bdd6f56769467853fd0242/src/bun_js.zig#L47-L66


This is where we call the code to evaluate the ESM module: https://github.com/oven-sh/bun/blob/6f59bc7d2e7aa64956bdd6f56769467853fd0242/src/bun.js/javascript.zig#L1537

This function is where it does the C++ equivalent of `import(path)`

https://github.com/oven-sh/bun/blob/6f59bc7d2e7aa64956bdd6f56769467853fd0242/src/bun.js/bindings/bindings.cpp#L2048-L2065
